packages/frontend/component/src/hooks/use-theme-color-meta.ts

```diff
@@ -0,0 +1,45 @@
+import { useEffect } from 'react';
+
+import { useThemeValueV1, useThemeValueV2 } from './use-theme-value';
+
+let meta: HTMLMetaElement | null = null;
+
+function getMeta() {
+  if (meta) return meta;
+
+  const exists = document.querySelector('meta[name="theme-color"]');
+  if (exists) {
+    meta = exists as HTMLMetaElement;
+    return meta;
+  }
+
+  // create and append meta
+  meta = document.createElement('meta');
+  meta.name = 'theme-color';
+  document.head.append(meta);
+  return meta;
+}
+
+export const useThemeColorMeta = (color: string) => {
```

Comment:
what is the use case?

Reply:
Used for `useThemeColorV1()`/ `useThemeColorV2()` to change browser's header color dynamically for different page or different state.

Reply:
it seems next-themes already support adding meta tag https://github.com/pacocoursey/next-themes/pull/114

Reply:
it seems next-themes already support adding meta tag https://github.com/pacocoursey/next-themes/pull/114
Ops it is only available for beta version of next-themes

Reply:
We need to change `theme-color` for different routes, and even different state like "when modal is opened"